src/OnePlusBot/Base/CommandHandler.cs

```diff
@@ -32,11 +32,60 @@ public CommandHandler(DiscordSocketClient bot, CommandService commands, IService
         public async Task InstallCommandsAsync()
         {
             _bot.MessageReceived += OnCommandReceived;
-            _bot.MessageReceived += OnMessageReceived;  
+            _bot.MessageReceived += OnMessageReceived;
+            _bot.MessageDeleted += OnMessageRemoved;
+            _bot.MessageUpdated += OnMessageUpdatedAsync;
             _commands.CommandExecuted += OnCommandExecutedAsync;
             await _commands.AddModulesAsync(Assembly.GetEntryAssembly(), _services);
         }
 
+        private async Task OnMessageUpdatedAsync(Cacheable<IMessage, ulong> cacheable, SocketMessage message, ISocketMessageChannel socketChannel)
+        {
+            var channel = socketChannel as SocketTextChannel;
+            var before = await cacheable.GetOrDownloadAsync();
+            var author = before.Author;
+
+            if (
+                before.Author == _bot.CurrentUser || message.Author == _bot.CurrentUser ||
+                before.Content == "" || message.Content == ""
+               )
+                return;
+
+            var embed = new EmbedBuilder
+            {
+                Color = Color.Blue,
+                Description = $":bulb: Message from '{author.Username}' edited in {channel.Mention}",
+                Fields = {
+                    new EmbedFieldBuilder() { IsInline = false, Name = $":x: Original message: ", Value = before.Content },
+                    new EmbedFieldBuilder() { IsInline = false, Name = $":pencil2: New message: ", Value = message.Content }
+                },
+                ThumbnailUrl = author.GetAvatarUrl(),
```

Comment:
Use often use Extension.RealAvatarUrl even tho I don't know if it really makes any difference

Reply:
Should be tested in the reports, might fix the invisible avatar.